tap_zendesk/streams.py

```diff
@@ -639,6 +642,49 @@ def check_access(self):
         '''
         self.client.sla_policies()
 
+class TalkPhoneNumbers(Stream):
+    name = 'talk_phone_numbers'
+    replication_method = "FULL_TABLE"
+    endpoint = 'https://{}.zendesk.com/api/v2/channels/voice/phone_numbers'
+
+    def sync(self, state): # pylint: disable=unused-argument
+        for phone_number in self.client.talk.phone_numbers():
+            yield (self.stream, phone_number)
+
+
+class TalkCalls(CursorBasedExportStream):
+    name = 'talk_calls'
+    replication_method = "INCREMENTAL"
+    replication_key = 'updated_at'
+    endpoint = 'https://{}.zendesk.com/api/v2/channels/voice/stats/incremental/calls'
+
+    def sync(self, state):
+        bookmark = self.get_bookmark(state)
+
+        if not bookmark:
+            start_date = self.config.get('start_date')
+            if start_date:
+                bookmark = utils.strptime_with_tz(start_date)
+
+        LOGGER.info(f'Start date: {bookmark}')
+
+        for call in self.client.talk.calls.incremental(start_time=bookmark):
+            if utils.strptime_with_tz(call.updated_at) >= bookmark:
+                self.update_bookmark(state, call.updated_at)
+                yield (self.stream, call)
+
+    def check_access(self):
+        '''
+        Check whether the permission was given to access stream resources or not.
+        '''
+        url = self.endpoint.format(self.config['subdomain'])
+        # Convert start_date parameter to timestamp to pass with request param
+        start_time = datetime.datetime.strptime(self.config['start_date'], START_DATE_FORMAT).timestamp()
+        HEADERS['Authorization'] = 'Bearer {}'.format(self.config["access_token"])
+
+        http.call_api(url, self.request_timeout, params={'start_time': f'{start_time:.0f}', 'per_page': 1}, headers=HEADERS)
```

Comment:
As we are bypassing 404 error and we are doing simplest call so we can refactor this like below: 
```
def check_access(self):
        try:
            epoch_start = int(utils.now().strftime('%s'))
            self.client.tickets.metrics_incremental(start_time=epoch_start)
        except http.ZendeskNotFoundError:
            #Skip 404 ZendeskNotFoundError error as goal is just to check whether TicketComments have read permission or not
            pass
 ```

Reply:
@dbshah1212 Actually, I copied the code from [here](https://github.com/singer-io/tap-zendesk/blob/master/tap_zendesk/streams.py#L379-L388). Shall this be changed then as well?

Or should that be maybe its own PR?

Reply:
Yes, This will also work. the approach I suggested was the latest but as we have your approach in many calls we can keep this. Resolving this comment.